DeepSpeech.ipynb

```diff
@@ -1320,170 +1292,390 @@
+    "        \n",
+    "        # Add batch to loss\n",
+    "        total_loss += result[offset]\n",
+    "        offset = offset + 1\n",
+    "        \n",
+    "        if do_training:\n",
+    "            # Log all variable states in current step\n",
+    "            step = epoch * data_set.total_batches + batch * len(available_devices)\n",
+    "            writer.add_summary(result[offset][1], step)\n",
+    "            if do_fulltrace:\n",
+    "                writer.add_run_metadata(loss_run_metadata, 'loss_epoch%d_batch%d'   % (epoch, batch))\n",
+    "            writer.flush()\n",
+    "            offset = offset + 1\n",
     "\n",
-    "    # Init all variables in session\n",
-    "    session.run(tf.initialize_all_variables())\n",
+    "        if query_report:\n",
+    "            # Collect individual sample results\n",
+    "            collect_results(report_results, result[offset][0])\n",
+    "            # Add batch to total_accuracy\n",
+    "            total_accuracy += result[offset][1]\n",
+    "            \n",
+    "    # Returning the batch set result tuple\n",
+    "    loss = total_loss / batches_per_device\n",
+    "    if query_report:\n",
+    "        return (loss, total_accuracy / batches_per_device, report_results)\n",
+    "    else:\n",
+    "        return (loss, None, None)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following routine will print a report from a provided batch set result tuple. It takes a caption for titling the output plus the batch set result tuple. If the batch set result tuple contains accuracy and a report results tuple, a complete WER report will be calculated, printed and its mean WER returned. Otherwise it will just print the loss and return `None`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def print_report(caption, batch_set_result):\n",
+    "    \n",
+    "    # Unpacking batch set result tuple\n",
+    "    loss, accuracy, results_tuple = batch_set_result\n",
     "    \n",
-    "    # Start queue runner threads\n",
-    "    tf.train.start_queue_runners(sess=session)\n",
+    "    # We always have a loss value\n",
+    "    title = caption + \" loss=\" + \"{:.9f}\".format(loss)\n",
+    "    \n",
+    "    mean_wer = None\n",
+    "    if accuracy is not None and results_tuple is not None:\n",
+    "        title += \" avg_cer=\" + \"{:.9f}\".format(accuracy)\n",
+    "        mean_wer = calculate_and_print_wer_report(title, results_tuple)\n",
+    "    else:\n",
+    "        print title\n",
+    "\n",
+    "    return mean_wer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's also introduce a routine that facilitates a complete run of a batch set (given by its name in `set_name`) - from execution contet creation to closing the session. \n",
```

Comment:
But you still can't 'run' a data-set, right? You perform things with/on it - I think we should be more explicit here so it can be understood in isolation.